### ISSUE TITLE
aggregateWithDeleted returns no result with Discriminators

### ISSUE THREAD

**Issue by davellx** [dsanel/mongoose-delete#130]
Hello,

I just had a problem while using aggregateWithDeleted (but I think it happens with aggregateDeleted method too)  that returned no results. 
I watched the resulting request and here's what I've found :
```javascript
[
   {
      "$match":{
         "deleted":{
            "$ne":true
         }
      }
   },
   {
      "$match":{
         "showAllDocuments":"true",
         "type":"rendezVousJuridique"
      }
   },
   // ... the rest of my pipeline
]
```
That's weird but I think the problem comes from the fact that my model is a Discriminator that also adds the key to the first found element. (I don't known exaclty ho mongoose does that so it's just a presumtion).

I can circumvent this by using the baseModel and adding a $match step in my pipeline but I think on this function:
https://github.com/dsanel/mongoose-delete/blob/2ea82825790ec9767f25f7c8df3cdeb74754ec6f/index.js#L139
It would maybe be better to use the objects to compare data with keys for conditions. It would be a little more greedy on cpu - not that much - but more accurate.

I don't have the time at the moment to make a PR but in a few weeks, if you want, I can try that.

**Comment by dsanel** [comment]
Hi @davellx did you have time to check this and maybe some code for PR? Thx

**Comment by davellx** [comment]
Finally I had some time today so here it is : 
https://github.com/dsanel/mongoose-delete/pull/135

**Comment by dsanel** [comment]
Merged and deployed in version [1.0.1](https://www.npmjs.com/package/mongoose-delete)

Thx @davellx
